Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertGeometry.cs

```diff
@@ -468,18 +469,37 @@ public CurveArray PolylineToNative(Polyline polyline)
       }
       else
       {
-        var pts = polyline.points;
-
+        var pts = polyline.GetPoints();
+        var lastPt = pts[0];
         for (var i = 1; i < pts.Count; i++)
         {
-          var speckleLine = new Line(new double[] { pts[i - 1].x, pts[i - 1].y, pts[i - 1].z, pts[i].x, pts[i].y, pts[i].z }, polyline.units);
-          curveArray.Append(LineToNative(speckleLine));
+          var speckleLine = new Line(lastPt, pts[i] , polyline.units);
+          var scaleToNative = ScaleToNative(speckleLine.length, speckleLine.units);
+          if (scaleToNative < Doc.Application.ShortCurveTolerance)
+            continue;
+          lastPt = pts[i];
+          try
+          {
+            var lineToNative = LineToNative(speckleLine);
+            curveArray.Append(lineToNative);
+          }
+          catch (Exception e)
+          {
+            Report.LogConversionError(e);
+          }
         }
 
         if (polyline.closed)
         {
-          var speckleLine = new Line(new double[] { pts[pts.Count - 1].x, pts[pts.Count - 1].y, pts[pts.Count - 1].z, pts[0].x, pts[0].y, pts[0].z }, polyline.units);
-          curveArray.Append(LineToNative(speckleLine));
+          var speckleLine = new Line(pts[pts.Count - 1], pts[0] , polyline.units);
```

Comment:
Should handle the case here where the polyline first and last points are the same and `.closed() = true`
The new Line may throw an exception in this case?

Reply:
fair, tho in theory this shouldn't happen ((the polyline class asks for first and last points to be unique tho this is not enforced so someone could easily do this if they don't read the docstring))